source/common/http/header_map_impl.cc

```diff
@@ -150,6 +150,26 @@ void HeaderString::append(const char* data, uint32_t size) {
   string_length_ += size;
 }
 
+void HeaderString::rtrim() {
+  switch (type_) {
+  case Type::Reference: {
+    RELEASE_ASSERT("false", "rtrim does not support header string in reference type.");
+  }
+  case Type::Dynamic: {
+    // rtrim only manipulate length_. Share the same code with Inline.
+    FALLTHRU;
+  }
+  case Type::Inline: {
+    absl::string_view original = getStringView();
+    absl::string_view rtrimmed = StringUtil::rtrim(original);
+    if (original.size() != rtrimmed.size()) {
+      string_length_ = rtrimmed.size();
+    }
+    break;
+  }
+  }
+}
+
```

Comment:
I think this can simply be:

void HeaderString::rtrim() {
  ASSERT(type() == Type::Inline || type() == Type::Dynamic);
  absl::string_view original = getStringView();
  absl::string_view rtrimmed = StringUtil::rtrim(original);
  if (original.size() != rtrimmed.size()) {
    string_length_ = rtrimmed.size();
  }
}

Reply:
That is closed to the original commit which I should follow.
Will update!